src/ro/catalog/Catalog.java

```diff
@@ -0,0 +1,110 @@
+package ro.catalog;
+
+import ro.catalog.utilizatori.Student;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+public class Catalog {
+    private List<Materie> materii = new ArrayList<>();
+    private List<Student> studenti = new ArrayList<>();
+    private int grupa;
+    private List<List<Integer>> note;
+
+    public Catalog(int grupa) {
+        this.grupa = grupa;
+    }
+
+    public void adaugareMaterie(Materie materie){
+        materii.add(materie);
+    }
+
+    public void adaugareStudent(Student student){
+        if(student.getGrupa() == grupa) {
+            studenti.add(student);
+        }
+    }
+
+    public int getMateriiSize(){
+        return materii.size();
+    }
+
+    public int getStudentiSize(){
+
+        return studenti.size();
+    }
```

Comment:
all these methods hold some logic and logic usually belongs to a service specific to a class - for example Catalog is an entity class, has 3 fields with setters getters constructor, toString, hashCode, equals, and in CatalogService you can define the methods for handling a Catalog, the ones you defined here. THis is a best practice for decoupling code